### Importing Libraries

In [1]:
import pandas as pd
import numpy as np

### Inspecting DataFrame

In [13]:
# load data
heart = pd.read_csv('heart_disease.csv')
heart.head()

age     sex  trestbps   chol                cp  exang  fbs  thalach  \
0  63.0    male     145.0  233.0    typical angina    0.0  1.0    150.0   
1  67.0    male     160.0  286.0      asymptomatic    1.0  0.0    108.0   
2  67.0    male     120.0  229.0      asymptomatic    1.0  0.0    129.0   
3  37.0    male     130.0  250.0  non-anginal pain    0.0  0.0    187.0   
4  41.0  female     130.0  204.0   atypical angina    0.0  0.0    172.0   

  heart_disease  
0       absence  
1      presence  
2      presence  
3       absence  
4       absence

### Filtering for patitents with heart disease

In [14]:
yes_hd = heart[heart.heart_disease == 'presence']
yes_hd.head()

age     sex  trestbps   chol            cp  exang  fbs  thalach  \
1  67.0    male     160.0  286.0  asymptomatic    1.0  0.0    108.0   
2  67.0    male     120.0  229.0  asymptomatic    1.0  0.0    129.0   
6  62.0  female     140.0  268.0  asymptomatic    0.0  0.0    160.0   
8  63.0    male     130.0  254.0  asymptomatic    0.0  0.0    147.0   
9  53.0    male     140.0  203.0  asymptomatic    1.0  1.0    155.0   

  heart_disease  
1      presence  
2      presence  
6      presence  
8      presence  
9      presence

### Filtering for patitents with no heart disease

In [15]:
no_hd = heart[heart.heart_disease == 'absence']
no_hd.head()

age     sex  trestbps   chol                cp  exang  fbs  thalach  \
0  63.0    male     145.0  233.0    typical angina    0.0  1.0    150.0   
3  37.0    male     130.0  250.0  non-anginal pain    0.0  0.0    187.0   
4  41.0  female     130.0  204.0   atypical angina    0.0  0.0    172.0   
5  56.0    male     120.0  236.0   atypical angina    0.0  0.0    178.0   
7  57.0  female     120.0  354.0      asymptomatic    1.0  0.0    163.0   

  heart_disease  
0       absence  
3       absence  
4       absence  
5       absence  
7       absence

### Mean cholestrol of patients with heart disease

In [16]:
chol_hd = yes_hd.chol
chol_hd.mean()

251.4748201438849

### Mean cholestrol of patients with no heart disease

In [17]:
chol_no_hd = no_hd.chol
chol_no_hd.mean()

242.640243902439

### Hypothesis to check if people with heart disease have high cholesterol levels (greater than or equal to 240 mg/dl) on average

In [19]:
from scipy.stats import ttest_1samp
tstat, pval = ttest_1samp(chol_hd, 240, alternative = 'greater')
pval

0.0035411033905155707

### Hypothesis to check if people without heart disease have high cholesterol levels (greater than or equal to 240 mg/dl) on average

In [20]:
tstat, pval = ttest_1samp(chol_no_hd, 240, alternative = 'greater')
pval

0.26397120232220506

### Number of patients

In [21]:
num_patients = len(heart)
num_patients

303

### Number of patients with fasting blood sugar greater than 120

In [22]:
num_highfbs_patients = len(heart[heart.fbs == 1])
num_highfbs_patients

45

### 8% of Sample Size

In [24]:
0.08 * num_patients

24.240000000000002

### Hypothesis test to check if this sample come from a population in which the rate of fbs > 120 mg/dl is equal to 8%

In [26]:
from scipy.stats import binom_test
p_value = binom_test(num_highfbs_patients, num_patients, 0.08, alternative = 'greater')
p_value

4.689471951449078e-05